pkg/executor/delete.go

```diff
@@ -55,10 +55,19 @@ type DeleteExec struct {
 // Next implements the Executor Next interface.
 func (e *DeleteExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	req.Reset()
+	var err error
 	if e.IsMultiTable {
-		return e.deleteMultiTablesByChunk(ctx)
+		err = e.deleteMultiTablesByChunk(ctx)
+	} else {
+		err = e.deleteSingleTableByChunk(ctx)
 	}
-	return e.deleteSingleTableByChunk(ctx)
+	if err != nil {
+		return err
+	}
+	if txn, _ := e.Ctx().Txn(false); txn != nil {
+		return txn.GetMemBuffer().MayFlush()
```

Comment:
Too late to call `MayFlush` here? I think you may need move `MayFlush` into `deleteSingleTableByChunk` and `deleteMultiTablesByChunk`.

Reply:
Good catch.